### PR TITLE
SyncBot: Support Sync within the same service

### PR DESCRIPTION
**Pull request by sqweelygig**
The observant in the audience will notice that this doesn't actually configure the OmniFlow.  I want to have this code in production for at least a few days before turning on the firehose.
If SyncBot is ever set up to Sync a flow which is not public, this will reveal information to other Resineers

* Track which flow a message originated from
* Allow messages within the same service, but different flows, to Sync.

Change-Type: patch

---
##### Contributor checklist

- [x] Build output been rebuilt and tested
- [x] Introduces security considerations
- [ ] Tests are included
- [ ] Documentation is added or changed
- [ ] Affects the development, build or deployment processes of the component

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: tests/services/messenger/translators/discourse.spec.ts [hunk 1]
```diff
@@ -2,14 +2,16 @@
 import { expect } from 'chai';
 import * as _ from 'lodash';
 
+import { BasicMessageInformation } from '../../../../lib/services/messenger-types';
```
Comment (LucianBuzzo) [inline tests/services/messenger/translators/discourse.spec.ts]: Using `paths` in you tsconfig can help with these really long file paths --> https://github.com/resin-io-modules/rendition/blob/master/tsconfig.json#L18

Comment (sqweelygig) [reply tests/services/messenger/translators/discourse.spec.ts]: @LucianBuzzo I've just skim read the rendition test files, and note that they all use the relative paths.  I'll be continuing to google and stack overflow about this, but wondered if your brain would be quicker.

File: package.json [hunk 2]
```diff
@@ -61,6 +61,7 @@
     "typescript": "~2.7.1"
   },
   "dependencies": {
+    "@types/bluebird": "^3.5.20",
```
Comment (sqweelygig) [inline package.json]: Note to self: wrong

File: lib/bots/syncbot.ts [hunk 3]
```diff
@@ -128,7 +128,7 @@ export class SyncBot extends ProcBot {
 			if (
 				from.service === data.source.service &&
 				from.flow === data.source.flow &&
-				to.service !== data.details.genesis &&
+				(to.service !== data.details.service || to.flow !== data.details.flow) &&
```
Comment (hedss) [inline lib/bots/syncbot.ts]: Ideally parenthesised within.

File: lib/services/messenger/translators/flowdock.ts [hunk 4]
```diff
@@ -63,7 +63,7 @@ export class FlowdockTranslator extends TranslatorScaffold implements Translator
 		const superFormat = (format === MetadataEncoding.Flowdock) ? MetadataEncoding.HiddenMD : format;
 		const metadata = TranslatorScaffold.extractMetadata(message, superFormat, config);
 		const findPublic = '^> *';
-		if (format === MetadataEncoding.Flowdock && metadata.genesis === null) {
+		if (format === MetadataEncoding.Flowdock && metadata.service === null) {
```
Comment (hedss) [inline lib/services/messenger/translators/flowdock.ts]: Parentheses.

**Pull request merged**